DoSomething/odoyle-scripts
# Use NorthstarId to load test broadcast instead of mobile

#### What's this PR do?
It uses a single configurable `northstarId` to load test broadcasts. All Requests will generate an import message for the same Northstar user.

#### How to test?
- Install locally.
- Set `DS_CONVERSATIONS_BROADCAST_SETTINGS_URL` to `<local|staging>/api/v1/broadcasts`.
- Set `DS_CONVERSATIONS_BROADCAST_ID` to `7zU0Mb1k9GkWWI40o06Mic`
- Set `DEFAULT_NORTHSTAR_ID` to your own staging northstarId
- cd into `/load-tests/broadcasts`
- Run `node index.js -u 1 -n 1 -s broadcast`
- You should get a text with the copy of the broadcast.

#### Notes
- This version of the loading test only works with the [new](https://github.com/DoSomething/gambit-conversations/pull/272) webhook payload from Gambit Conversations `/broadcasts/:id`.
    ```
    {
        "northstarId": "{{customer.id}}",
        "broadcastId": "7zU0Mb1k9GkWWI40o06Mic"
    }
    ```
